src/include/daos_fs.h

```diff
@@ -98,6 +98,10 @@ typedef struct {
 	daos_oclass_id_t	doi_oclass_id;
 	/** chunk size */
 	daos_size_t		doi_chunk_size;
+	/** In case of dir, return default object class for dirs created in that dir */
+	daos_oclass_id_t        doi_dir_oclass_id;
+	/** In case of dir, return default object class for files created in that dir */
+	daos_oclass_id_t        doi_file_oclass_id;
```

Comment:
Do we care to make this backward compatible? E.g. if an application is built against 2.4.x, but then daos is updated and the app not rebuilt

Reply:
this is used in the daos tools and I'm not aware of other middleware or apps using that.
that being said, just adding these fields at the end of the struct won't cause issues. the main issue is the change in behavior of what doi_oclass_id is for dirs. but as i said this is just used today by DAOS itself.

Reply:
> just adding these fields at the end of the struct won't cause issues.

I think it would if an application was compiled against an older version and, e.g., the size of the struct was 1K but now with the additional fields the size is 2K. So when the application initializes an instance of the struct, it uses too small of a size. Then when dfs tries to access the struct members it's a memory error.

IIRC, we ran into a similar issue before. Anyway, I'll defer to your judgement

Reply:
yea sorry i was thinking compiling issues. you are right though.
but im not worried about it as this was mainly written for the tools.

Reply:
Probably safe for now.   As we use more of these management APIs though, we will probalby need to be more careful and bump major/minor API versions

Reply:
Yea i thought about bumping the API major version, but it would be an overkill in this case because of how this is used today. a minor version bump probably makes more sense, but i will evaluate that with 2.6 release to see if it was already done or not to avoid bumping it multiple times.

Reply:
APIs/ABIs can be tricky but extending structs is one of the few things that you can get away with.  Doing a scan for API (header) changes and ensuring appropriate major/minor bumps wouldn't be a bad thing to have on the TODO list at the beginning of a release cycle.